### ISSUE TITLE
Import metadata from EXIF

### ISSUE THREAD

**Issue by pierre-guillot** [pixelfed/pixelfed#499]
I'll love to have exif details about my picture each time I upload a new one one pixelfed. Details about used camera, aperture, speed etc. 
Of course, picture needs exif on it, and this functionality must be enabled on account option for work (privacy reason).

**Comment by carbontwelve** [comment]
I'd also like to see the option of having pixelfed strip EXIF meta from uploaded images, for the purpose of improving privacy e.g removing geolocation (or have the ability to anonymise it in a 50mi radius)

**Comment by peterjw** [comment]
@carbontwelve 

> I'd also like to see the option of having pixelfed strip EXIF meta from uploaded images, for the purpose of improving privacy e.g removing geolocation (or have the ability to anonymise it in a 50mi radius)

To my experience, this is the current default - isn't it?

**Comment by carbontwelve** [comment]
@peterjw 

> To my experience, this is the current default - isn't it?

You're probably right.

**Comment by maznu** [comment]
There's a huge debate raging between photographers and publishers at the moment. Within the last couple of weeks Google, Facebook, and other "platforms" have stopped stripping _all_ EXIF and IPTC tags from images. Various news publications (whose tech teams run everything through imagemagick/other pipelines and strip everything) have also become involved. The problem they're all addressing is, ultimately, the loss of revenue due to image (copyright) theft. It's too easy for someone to find a photo in an image search and assume they "can just use it" (this is a big education problem, for sure).  To help educate users, for example, [Google Image Search is now starting to display copyright/ownership metadata](https://www.forbes.com/sites/paulmonckton/2018/09/30/new-google-image-search-feature-will-please-photographers/#67a5111c2074) in its results as a sort of tooltip.

Another thing that has happened in the last few days is [Flickr's change in business model](https://www.flickr.com/lookingahead). I suspect there will be a lot of photographers looking for a new home, and maybe this could be the fediverse.  Pixelfed takes a lot of inspiration from Instagram's user experience… but maybe there will be "prosumer" instances which riff off some of the features that Flickr offers?

Rather than stripping all tags and metadata — which is laudable from a privacy perspective, protecting users from inadvertent disclosures having uploaded GPS-tagged photos taken within/around their home — I propose:

* never strip IPTC copyright/description/ownership/license tags (reducing "orphan works" or excuses from publications not seeking out authors of copyright works)
* option to include camera shot EXIF data (but maybe the choice to hide camera/lens models, or serial numbers, etc)
* option to include GPS data (and/or an option to make it more "fuzzy") — a feature which would compete with Flickr's mapping information and allow a future map-centric view, or geo-* feeds of images
* include some sort of UI showing the technical information about a photograph (it's just one of many ways to learn technique)

Wouldn't it be amazing if I could just upload my photos straight out of e.g. Lightroom, and Pixelfed prepopulated the image post using the captions, copyright, and other metadata available to it?

**Comment by toastal** [comment]
I'd like to reiterate @maznu as a person looking at alternatives to Flickr...

- Currently with `strip-all` you're tearing off ICC profiles. I don't want to lose my color accuracy on upload, and I don't want users that have better-than-sRGB monitors to have to see worse colors.
- Keeping copyright is good for users who don't want their images stolen, and the tags, title, etc. give the uploader some content to work with. Turn all tags into hashtags and now you offer a better user experience than Instagram while mimicking the features of Flickr. Even Facebook reads the title if present to help construct the body for the photo.
- Being able to search on camera data is super, super crucial as there's not a lot of places to see what my new $15,832 lens will look like with my camera body.
- Additionally can also be used to promote FOSS tools like give the photo a badge if it was edited in darktable, RAWTherapee, GIMP, et. al.

Stripping EXIF data should be optional for privacy and file size (and longer-term configurable), but since this is a photo platform, photography geek stats are must if you intend to attract  loyal users and be more competitive outside of "it's the fediverse". Because my attribution, Creative Commons licenses, and color profiles are being stripped, PixelFed is a no-go for me.

**Comment by dansup** [comment]
This will be possible with the new compose UI after some more commits are pushed!

**Comment by carbontwelve** [comment]
@maznu 
> Wouldn't it be amazing if I could just upload my photos straight out of e.g. Lightroom, and Pixelfed prepopulated the image post using the captions, copyright, and other metadata available to it?

Yes, for sure 👍when I mentioned about stripping EXIF tags I meant those relating to privacy such as GPS - or at the very least anonymising the location within a 100mi radius of the origin.

**Comment by trwnh** [comment]
Would also be nice to see filtering by the preserved metadata? Something like a "Shot with [camera]" search would be cool.

**Comment by maznu** [comment]
Looks like the `media.metadata` field has got some good stuff in.

But I'm still concerned about the stripping of all the caption, copyright, and other metadata from the JPEGs that are created and served to the end-user.  As per my earlier comment, retaining this information is crucial for third parties (especially search engines and copyright thieving tabloid newspapers) to know the usage rights of an image file.
